src/__tests__/index.jsx

```diff
@@ -68,4 +82,84 @@ describe('<ReactTimeline />', () => {
     wrapper.setProps(nextProps)
     expect(wrapper.state('time').zoom).toBe(2)
   })
+
+  describe('sticky header', () => {
+    it('is sticky when top of window is within the timeline', () => {
+      const listeners = {}
+      addListener.mockImplementation((evt, fun) => { listeners[evt] = fun })
+      removeListener.mockImplementation(jest.fn())
+      raf.mockImplementation(fn => fn())
+
+      getComputedStyle.mockImplementation(node => ({
+        getPropertyValue(prop) {
+          return node.style[prop]
+        }
+      }))
+
+      const props = createProps({ stickyHeader: true })
+      const wrapper = mount(<ReactTimeline {...props} />)
+      expect(typeof listeners.scroll).toEqual('function')
+
+      wrapper.instance().getMarkerOffset({
+        style: { 'padding-top': '40px' }
+      })
+      wrapper.instance().getHeaderHeight({
+        offsetHeight: 50
+      })
+      wrapper.instance().layoutMain.getBoundingClientRect = () => ({
+        top: -60,
+        bottom: 100
+      })
+      listeners.scroll()
+      expect(wrapper.state()).toMatchObject({
+        isHeaderSticky: true
+      })
+
+      wrapper.instance().layoutMain.getBoundingClientRect = () => ({
+        top: 10,
+        bottom: 100
+      })
+      listeners.scroll()
+      expect(wrapper.state()).toMatchObject({
+        isHeaderSticky: false
+      })
+
+      wrapper.instance().layoutMain.getBoundingClientRect = () => ({
+        top: -60,
+        bottom: 20
+      })
+      listeners.scroll()
+      expect(wrapper.state()).toMatchObject({
+        isHeaderSticky: false
+      })
+
+      wrapper.unmount()
+      expect(removeListener).toBeCalled()
```

Comment:
You're probably right but I'll want some feedback from @WebSeed before I do this as I was testing the addition of the event listeners on mount before being encourage to write more BDD style tests.